avocado/core/nrunner.py

```diff
@@ -76,26 +76,28 @@ class Runnable:
     A instance of :class:`BaseRunner` is the entity that will actually
     execute a runnable.
     """
-    def __init__(self, kind, uri, *args, **kwargs):
+    def __init__(self, kind, uri, config, *args, **kwargs):
```

Comment:
Sorry that I also missed this earlier, but AFAICT there's no need for the `config` to be mandatory, right?  Something like this:

```diff
diff --git a/avocado/core/nrunner.py b/avocado/core/nrunner.py
index 76322aba3..d4b3213e1 100755
--- a/avocado/core/nrunner.py
+++ b/avocado/core/nrunner.py
@@ -76,9 +76,11 @@ class Runnable:
     A instance of :class:`BaseRunner` is the entity that will actually
     execute a runnable.
     """
-    def __init__(self, kind, uri, config, *args, **kwargs):
+    def __init__(self, kind, uri, config=None, *args, **kwargs):
         self.kind = kind
         self.uri = uri
+        if config is None:
+            config = {}
         self.config = config
         self.args = args
         self.tags = kwargs.pop('tags', None)
```

Should be enough.

Reply:
+1 here too!

Reply:
You are right, It was on my todo list and I completly forgot that.